appium/webdriver/extensions/gsm.py

```diff
@@ -16,39 +16,50 @@
 from ..mobilecommand import MobileCommand as Command
 
 from appium.common.logger import logger
+from ..gsm_signal_strength import GsmSignalStrength
+from ..gsm_call_actions import GsmCallActions
 
 
 class Gsm(webdriver.Remote):
 
-    (
-        NONE_OR_UNKNOWN,
-        POOR,
-        MODERATE,
-        GOOD,
-        GREAT
-    ) = range(5)
+    def make_gsm_call(self, phone_number, action):
+        """Make GSM call (Emulator only)
+
+        :Args:
+         - phone_number (str): The phone number to call to.
+         - action (str): The action - GsmCallActions.CALL/ACCEPT/CANCEL/HOLD
+
+        :Usage:
+            self.driver.make_gsm_call('5551234567', GsmCallActions.CALL)
+        """
+        if action not in self._dict_const(GsmCallActions).values():
+            logger.warning('{} is invalid. Use the value in {}. (e.g. GsmCallActions.CALL)'.format(
+                action, list(self._dict_const(GsmCallActions).keys())))
+        self.execute(Command.MAKE_GSM_CALL, {'phoneNumber': phone_number, 'action': action})
+        return self
 
     def set_gsm_signal(self, strength):
         """Set GSM signal strength (Emulator only)
 
         :Args:
-         - strength: Signal strength. Can be set Gsm.NONE_OR_UNKNOWN/POOR/MODERATE/GOOD/GREAT
+         - strength: Signal strength. Can be set GsmSignalStrength.NONE_OR_UNKNOWN/POOR/MODERATE/GOOD/GREAT
 
         :Usage:
-            self.driver.set_gsm_signal(Gsm.GOOD)
+            self.driver.set_gsm_signal(GsmSignalStrength.GOOD)
         """
-        if strength not in self._dict_signal_strength().values():
-            logger.warning('{} is out of range. Use the value in {}.'.format(
-                strength, list(self._dict_signal_strength().keys())))
+        if strength not in self._dict_const(GsmSignalStrength).values():
+            logger.warning('{} is out of range. Use the value in {}. (e.g. GsmSignalStrength.GOOD)'.format(
+                strength, list(self._dict_const(GsmSignalStrength).keys())))
         self.execute(Command.SET_GSM_SIGNAL, {'signalStrength': strength, 'signalStrengh': strength})
         return self
 
-    def _dict_signal_strength(self):
-        return {'{}.{}'.format(Gsm.__name__, attr): value for attr, value in vars(Gsm).items()
-                if not callable(getattr(Gsm, attr)) and attr.isupper()}
+    def _dict_const(self, cls):
```

Comment:
I think this should not be a class method, but rather a helper function. Can we also have a better name for it?